misc/package/Data Files/MWSE/core/mcm/components/settings/Setting.lua

```diff
@@ -21,10 +23,38 @@ Setting.restartRequiredMessage = mwse.mcm.i18n("The game must be restarted befor
 function Setting:new(data)
 	local t = Parent:new(data)
 
-	if data and data.variable then
-		-- create setting variable
-		t.variable.defaultSetting = t.variable.defaultSetting or t.defaultSetting
-		t.variable = fileUtils.getVariableClass(t.variable.class):new(t.variable)
+	
+	if data then
+		local configKey = data.configKey
+		local parent = data.parentComponent
+		local config = data.config or parent and parent.config
+		local defaultConfig = data.defaultConfig or parent and parent.defaultConfig
+
+		-- Get the default setting. Include `nil` checks so we can handle it being `false`.
+		local defaultSetting = data.variable and data.variable.defaultSetting
+		if defaultSetting == nil then
+			defaultSetting = data.defaultSetting
+			if defaultSetting == nil and defaultConfig and configKey then
+				defaultSetting = defaultConfig[configKey]
+			end
+		end
```

Comment:
IMHO, could be a bit more readable as:
```lua
local defaultSetting = data.variable and data.variable.defaultSetting or data.defaultSetting
if defaultSetting == nil and defaultConfig and configKey then
		defaultSetting = defaultConfig[configKey]
end
```

Reply:
I would have preferred that as well, but the downside of that approach is that it has some weird behavior when `data.variable.defaultSetting == false` (e.g. if `data` is a `YesNoButton`). More specifically, if `data.defaultSetting == nil` and `data.variable.defaultSetting == false`, then 
```lua
local defaultSetting = data.variable and data.variable.defaultSetting or data.defaultSetting
```
will set `defaultSetting = nil`.

Reply:
Oh, right, it's not functionally identical. Still, my inner non-nester feels it's a bit nicer if it's done as:
```lua
local defaultSetting = data.variable and data.variable.defaultSetting
if defaultSetting == nil then
	defaultSetting = data.defaultSetting
end

if defaultSetting == nil and defaultConfig and configKey then
	defaultSetting = defaultConfig[configKey]
end
```
Feel free to ignore this suggestion of you don't find it more readable that way.

Reply:
I'll make this change. I agree that it's more readable this way. It's always nice to cut down on nesting whenever possible.